code/go/0chain.net/smartcontract/stakepool/stakepool.go

```diff
@@ -52,6 +53,17 @@ func NewStakePool() *StakePool {
 	}
 }
 
+func (sp *StakePool) Copy() *StakePool {
+	spCopy := NewStakePool()
+	for k, v := range sp.Pools {
+		spCopy.Pools[k] = v
```

Comment:
This would only copy the pointers, so it could be changed, please copy the value instead of pointer.